mono/metadata/assembly.c

```diff
@@ -2168,14 +2168,12 @@ mono_assembly_open_from_bundle (const char *filename, MonoImageOpenStatus *statu
 	is_satellite = g_str_has_suffix (lowercase_filename, ".resources.dll");
 	g_free (lowercase_filename);
 	name = g_path_get_basename (filename);
-	mono_assemblies_lock ();
 	for (i = 0; !image && bundles [i]; ++i) {
 		if (strcmp (bundles [i]->name, is_satellite ? filename : name) == 0) {
 			image = mono_image_open_from_data_internal ((char*)bundles [i]->data, bundles [i]->size, FALSE, status, refonly, FALSE, name);
```

Comment:
I think we should keep the lock, and call the open function outside the lock.

Reply:
Any specific reasons that comes to mind keeping it? Doing unlock/lock around mono_image_open_from_data_internal will not give us much more safety for the loop and access to bundles is not protected using a lock in other places in this source file.

Reply:
It looks like bundles are inited before runtime startup so no locking is needed.